vector/src/main/java/im/vector/app/features/pin/lockscreen/ui/LockScreenFragment.kt

```diff
@@ -55,22 +59,12 @@ class LockScreenFragment : VectorBaseFragment<FragmentLockScreenBinding>() {
             handleEvent(it)
         }
 
-        withState(viewModel) { state ->
-            if (state.lockScreenConfiguration.mode == LockScreenMode.CREATE) return@withState
-
-            viewLifecycleOwner.lifecycleScope.launchWhenResumed {
-                if (state.canUseBiometricAuth && state.isBiometricKeyInvalidated) {
-                    lockScreenListener?.onBiometricKeyInvalidated()
-                } else if (state.showBiometricPromptAutomatically) {
+        viewModel.stateFlow.distinctUntilChangedBy { it.showBiometricPromptAutomatically }
```

Comment:
I'm wondering if we should expose this as a `ViewEvent`, the `ViewModel.init` would register to the state and emit the event in `onEach` rather than the `Fragment` handling the logic (would have the benefit of being testable at the `ViewModel` layer)
 
what do you think? (could be a separate PR if you agree)

Reply:
I think we don't need to observe the field, just checking it once on `init`, inside the launched coroutine should be enough: we only want to display the prompt automatically once when the user enters the lockscreen, so triggering the event several times depending on external changes might be weird behaviour.

Reply:
I see! in that case `flow.first { it.showBiometricPromptAutomatically  }` might be better than distinct + filter 

it's also worth highlighting `init` is probably a bad place to emit events as it requires the fragments to be able to start observing before the `ViewModel` has been created, otherwise it creates race conditions :sweat_smile: 

other `ViewModels` has dedicated actions that correlate to the lifecycle like `Resumed/Visible`

Reply:
Do you mean something like this?

```kotlin
init {
    ...
    observeStateChanges()
}

private fun observeStateChanges() {
    viewModelScope.launch {
        if (stateFlow.firstOrNull { it.showBiometricPromptAutomatically } != null) {
            _viewEvents.post(LockScreenViewEvent.ShowBiometricPromptAutomatically)
        }
    }

    viewModelScope.launch {
        if (stateFlow.firstOrNull { it.isBiometricKeyInvalidated } != null) {
            onBiometricKeyInvalidated()
        }
    }
}
```

Reply:
yep ^^^ but if it's possible for the view event to trigger before a fragment has a chance to start observing we'll need another `ViewAction` when `onCreate` or `onResume` happens to trigger the logic rather instead of `init`

Reply:
I added an `OnUIReady` action to be triggered when the fragment's `onViewCreated` is called, right after it starts observing events.